modules/xray-udp-exporter/src/main/scala/io/janstenpickle/trace4cats/xray/XRayUdpSpan.scala

```diff
@@ -0,0 +1,167 @@
+package io.janstenpickle.trace4cats.xray
+
+import cats.{Applicative, Eval, Functor, Monad}
+import cats.data.NonEmptyList
+import cats.effect.{Clock, Ref}
+import cats.effect.std.Random
+import cats.syntax.applicative._
+import cats.syntax.apply._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import fs2.Chunk
+import io.circe.{Encoder, Json, JsonObject, Printer}
+import io.circe.syntax._
+import io.janstenpickle.trace4cats.`export`.SemanticTags
+import io.janstenpickle.trace4cats.model._
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import scala.util.matching.Regex
+
+private[xray] object XRayUdpSpan {
+
+  /** Span names can contain Unicode letters, numbers, and whitespace, and a limited set of symbols.
+    *
+    * See the `name` documentation under "Required Segment Fields"
+    * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-fields
+    */
+  private val nameRegex: Regex = """[^\p{L}0-9 _\.:/%&#=\+\-@]""".r
+
+  /** Annotation keys must be alphanumeric in order to work with filters. Underscore is allowed. Other symbols and
+    * whitespace are not allowed.
+    *
+    * See the documentation under "Annotations"
+    * https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-annotations
+    */
+  private val keyRegex: Regex = """[^A-Za-z0-9_]""".r
```

Comment:
```suggestion
  private val keyIllegalCharRegex: Regex = """[^A-Za-z0-9_]""".r
```

Reply:
Same as above, this regex will match all characters except the patterns listed here. How about this?

```suggestion
  private val keyAllowedCharRegex: Regex = """[^A-Za-z0-9_]""".r
```